src/main/java/org/terasology/launcher/util/Platform.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.util;
+
+/**
+ * A simplified representation of a computer platform as `os` and `arch`
+ */
+
+public final class Platform {
+    
+    private static final Platform PLATFORM = new Platform();
+
+    private String os;
+    private String arch;
+
+    /**
+     * Constructs platform information for the current host system.
+     * Simplifies operating system name to one of `linux`, `mac`, `windows` if applicable.
+     * Simplifies operating system architecture to one of `32` and `64` if applicable.
+     */
+    private Platform() {
+        final String platformOs = System.getProperty("os.name").toLowerCase();
+        // TODO: consider using regex
+        if (platformOs.startsWith("linux")) {
+            this.os = "linux";
```

Comment:
Using `this.` all the time is superfluous here. There is no need to use it as there is no ambiguity to what `os` is. Same holds for `arch`, here in the constructor and all the getters.

Reply:
Done